nssdb/generate/main.go

```diff
@@ -0,0 +1,197 @@
+//go:build ignore
+
+package main
+
+import (
+	"bufio"
+	"bytes"
+	_ "embed"
+	"fmt"
+	"log"
+	"os"
+	"regexp"
+	"strings"
+	"text/template"
+)
+
+const (
+	// Constants and flags used to generate the column number IDs.
+	// These were manually copied from lib/util/pkcs11t.h and pkcs11n.h.
+	NSSCK_VENDOR_NSS    = 0x4E534350
+	CKA_VENDOR_DEFINED  = 0x80000000
+	CKA_NSS             = CKA_VENDOR_DEFINED | NSSCK_VENDOR_NSS
+	CKA_TRUST           = CKA_NSS + 0x2000
+	CKA_SUBPRIME_BITS   = 0x00000134
+	CKF_ARRAY_ATTRIBUTE = 0x40000000
+)
+
+// This file contains the knownAttributes array, which is the list of columns.
+//
+//go:embed nss/lib/softoken/sdb.c
```

Comment:
We generally won't need to re-generate those files any time soon, but it would be nice to explain the requirements and how to set up the environment to regenerate `columns.go`.

I'm not sure about the nuances of MPL-2.0 and Apache, so I think it is better not to provide these files here, but we should not spend much time investigating how to make it work.

Reply:
I left some instructions in the README https://github.com/smallstep/crypto/pull/690/files#diff-4d575b5bcadf9b7387d6b3af5299f8614a0035fc09422b7bf5c41428eafd8eb3R44